src/quickpicks/cloudProviderRegions.ts

```diff
@@ -47,3 +109,90 @@ export async function cloudProviderRegionQuickPick(
 
   return chosenRegion?.value;
 }
+
+/**
+ * Quickpick that shows only the cloud provider + region combinations that the user can access
+ * through existing Flink databases (Flinkable Kafka clusters).
+ * Includes a trailing "View All" item which, if selected, opens the full list of region/providers
+ * {@link cloudProviderRegionQuickPick} (with optional filter parameter forwarded) so the user can
+ * choose among all available regions.
+ * @returns {Promise<IProviderRegion | undefined>} region selected or undefined if cancelled.
+ */
+export async function flinkDatabaseRegionsQuickPick(
+  filter?: regionFilter,
+): Promise<IProviderRegion | undefined> {
+  const loader = CCloudResourceLoader.getInstance();
+  const flinkDbClusters: CCloudFlinkDbKafkaCluster[] = await loader.getFlinkDatabases();
+
+  const awsSet = new ObjectSet<IProviderRegion>((pr) => `${pr.provider}-${pr.region}`);
+  const azureSet = new ObjectSet<IProviderRegion>((pr) => `${pr.provider}-${pr.region}`);
+  // GCP not supported for Flink artifacts yet
+  for (const db of flinkDbClusters) {
+    const pr = {
+      provider: db.provider,
+      region: db.region,
+      name: db.name,
+    };
+    if (db.provider === "AWS") {
+      awsSet.add(pr);
+    } else if (db.provider === "AZURE") {
+      azureSet.add(pr);
+    }
+  }
+  const awsProviderRegions: IProviderRegion[] = awsSet
+    .items()
+    .sort((a, b) => a.region.localeCompare(b.region));
+  const azureProviderRegions: IProviderRegion[] = azureSet
+    .items()
+    .sort((a, b) => a.region.localeCompare(b.region));
+
+  const quickPickItems: QuickPickItemWithValue<IProviderRegion | "VIEW_ALL">[] = [];
+  let lastProvider = "";
+  for (const entry of [...awsProviderRegions, ...azureProviderRegions]) {
+    if (entry.provider !== lastProvider) {
+      lastProvider = entry.provider;
+      quickPickItems.push({
+        label: entry.provider,
+        description: "",
+        kind: QuickPickItemKind.Separator,
+      });
+    }
+    // make the description out of the names of the databases in this provider+region
+    const matchingDatabases = flinkDbClusters.filter(
+      (db) => db.provider === entry.provider && db.region === entry.region,
+    );
+    quickPickItems.push({
+      label: `${entry.provider} | ${entry.region}`,
+      description: matchingDatabases.map((db) => db.name).join(", "),
+      value: { provider: entry.provider, region: entry.region },
+    });
+  }
+
+  quickPickItems.push(
+    {
+      label: "View All Available Regions",
```

Comment:
Minor: I don't think we need the label on the separator here - what are your thoughts there?

<img width="723" height="155" alt="Image" src="https://github.com/user-attachments/assets/3d85e682-f864-4066-98ed-6f4dc3d877c0" />